### PR TITLE
Add OAuth login / registration

### LINKED ISSUE #6: Add OAuth signup
> Should include option to use google or github

Django plugin: [`social-auth-app-django`](https://github.com/python-social-auth/social-app-django)

### PR DESCRIPTION
**Pull request by dulldesk**
Fixes #6

### CODE REVIEW

File: stratus/settings.py [hunk 1]
```diff
@@ -130,3 +137,8 @@
 
 LOGIN_REDIRECT_URL = '/home/'
 LOGIN_URL = '/login/'
+
+try:
+    from local_settings import *
+except ImportError:
+    pass
```
Comment (fluix-dev) [inline stratus/settings.py]:
```suggestion
```
People do this and I have mixed opinions. Django has a `--settings` for a reason, so let's use that.

File: app/urls.py [hunk 2]
```diff
@@ -7,6 +7,7 @@
     path("register/", views.RegisterView.as_view(), name="register"),
     path("login/", auth_views.LoginView.as_view(template_name="login.html"), name="login"),
     path("logout/", auth_views.LogoutView.as_view(template_name="logout.html"), name="logout"),
+    path("login/social/", include("social_django.urls", namespace="social")),
```
Comment (fluix-dev) [inline app/urls.py:13]: Move this one line up.

File: app/urls.py [hunk 3]
```diff
@@ -1,4 +1,4 @@
-from django.urls import path
+from django.urls import path, include
```
Comment (fluix-dev) [inline app/urls.py]: Switch order.

File: app/templates/register_login.html [hunk 4]
```diff
@@ -27,5 +27,24 @@
...
     </form>
+    <div class="frame__body form-section">
+      <label>Or {{ type|lower }} with:</label>
+      <div class="row">
+        <div class="col"> 
+          <a href="{% url "social:begin" "google-oauth2" %}" alt="{{ type }} with Google" title="Google">
+            <button href="{% url "social:begin" "google-oauth2" %}" class="btn-grey u-no-margin w-100" >
+              <i class="zmdi zmdi-google"></i>&nbsp;<span>Google</span>
+            </button>
+          </a>
```
Comment (fluix-dev) [inline app/templates/register_login.html]:
```suggestion
          <a href="{% url "social:begin" "google-oauth2" %}" class="btn-grey u-no-margin w-100">
              <i class="zmdi zmdi-google"></i> Google
          </a>
```
I'm pretty sure this looks the same.

File: app/templates/register_login.html [hunk 5]
```diff
@@ -27,5 +27,24 @@
...
+        </div>
+        <div class="col"> 
+          <a href="{% url "social:begin" "github" %}" alt="{{ type }} with GitHub" title="GitHub">
+            <button href="{% url "social:begin" "github" %}" class="btn-grey u-no-margin w-100" >
+              <i class="zmdi zmdi-github"></i>&nbsp;<span>GitHub</span>
+            </button>
+          </a>
```
Comment (fluix-dev) [inline app/templates/register_login.html]: Ditto.

File: app/static/css/style.css [hunk 6]
```diff
@@ -262,4 +262,16 @@ footer{
...
   top: 5px;  
-}
\ No newline at end of file
+}
+
+.btn-grey {
+  --btn-color: var(--grey-mid);
+  --btn-border-color: var(--grey-mid);
+  /*--btn-color: var(--cirrus-gray);*/
+  /*--btn-border-color: var(--cirrus-gray);*/
```
Comment (fluix-dev) [inline app/static/css/style.css]:
```suggestion
```

File: app/templates/register_login.html [hunk 7]
```diff
@@ -27,5 +27,24 @@
...
     </form>
+    <div class="frame__body form-section">
+      <label>Or {{ type|lower }} with:</label>
+      <div class="row">
+        <div class="col"> 
+          <a href="{% url "social:begin" "google-oauth2" %}?next={{ next }}" alt="{{ type }} with Google" title="Google">
+            <button class="btn-grey u-no-margin w-100">
+              <i class="zmdi zmdi-google"></i>&nbsp;<span>Google</span>
+            </button>
+          </a>
```
Comment (fluix-dev) [inline app/templates/register_login.html]:
```suggestion
          <a href="{% url "social:begin" "google-oauth2" %}?next={{ next }}" class="btn-grey u-no-margin w-100">
              <i class="zmdi zmdi-google"></i> Google
          </a>
```
I'm pretty sure this looks the same (maybe not the `<span>` removal), `alt` doesn't get used on `<a>`, and `title` doesn't serve any better purpose here.

File: app/templates/register_login.html [hunk 8]
```diff
@@ -27,5 +27,24 @@
...
+        </div>
+        <div class="col"> 
+          <a href="{% url "social:begin" "github" %}?next={{ next }}" alt="{{ type }} with GitHub" title="GitHub">
+            <button class="btn-grey u-no-margin w-100">
+              <i class="zmdi zmdi-github"></i>&nbsp;<span>GitHub</span>
+            </button>
+          </a>
```
Comment (fluix-dev) [inline app/templates/register_login.html]:
```suggestion
```
Ditto.

File: app/static/css/style.css [hunk 9]
```diff
@@ -184,7 +184,7 @@ footer{
 	color: var(--grey-mid);
 }
 
-.row [class^=col-]:not(.col-12){
+.row:not(.no-space) [class^=col-]:not(.col-12) {
```
Comment (fluix-dev) [inline app/static/css/style.css]: Is this related to this PR? I don't see `.no-space` used anywhere.

**Review by fluix-dev — CHANGES_REQUESTED** [review]
Otherwise seems nice and simple.

File: app/static/css/style.css (hunk 9 above)
Comment (dulldesk) [reply app/static/css/style.css]: It was

File: app/templates/register_login.html [hunk 10]
```diff
@@ -27,5 +27,20 @@
         <input type="submit" class="btn-primary u-pull-right animated" value="{{ type }}">
       </div>
     </form>
+    <div class="frame__body form-section">
+      <label>Or {{ type|lower }} with:</label>
+      <div class="row">
+        <div class="col"> 
+          <a href="{% url "social:begin" "google-oauth2" %}?next={{ next }}" class="btn btn-grey u-no-margin w-100">
```
Comment (fluix-dev) [inline app/templates/register_login.html]: Now that #33 has been merged, change this to Jinja url reversing.

File: app/templates/register_login.html [hunk 11]
```diff
@@ -27,5 +27,20 @@
...
+            <i class="zmdi zmdi-google"></i>&nbsp;<span>Google</span>
+          </a>
+        </div>
+        <div class="col"> 
+          <a href="{% url "social:begin" "github" %}?next={{ next }}" class="btn btn-grey u-no-margin w-100">
```
Comment (fluix-dev) [inline app/templates/register_login.html]: Ditto.

**Review by fluix-dev — CHANGES_REQUESTED** [review]
Last thing :smiley:

**Comment by dulldesk** [comment]
Should the login page be available to already-logged-in users? Currently, if signed-in users "re-sign in" using social authentication, the social authentication is connected to that account.

**Comment by fluix-dev** [comment]
Thanks!
